### PR TITLE
E-343 feat(utils): types definition generation

### PR DESCRIPTION
**Pull request by lamartire**

### CODE REVIEW

**Comment by lamartire** [comment]
@ifiri review this as draft

File: packages/utils/src/keystoreKeyVerify.js [hunk 1]
```diff
@@ -1,27 +1,22 @@
...
-      Buffer.from(value.replace(/^0x/, ''), 'hex'),
-    );
-  },
+/**
+ * Verify public key
+ * @param {string} value
+ * @returns {boolean}
+ */
+const verifyPublicKey = value =>
+  secp256k1.publicKeyVerify(Buffer.from(value.replace(/^0x/, ''), 'hex'));
```
Comment (ifiri) [inline packages/utils/src/keystoreKeyVerify.js]: In my opinion, previous variant was look more clean and readable. Is this prettier autofix or something manual?
```
const verifyPublicKey = value => secp256k1.publicKeyVerify(
  Buffer.from(value.replace(/^0x/, ''), 'hex')
);

// or
const verifyPublicKey = value => 
  secp256k1.publicKeyVerify(
    Buffer.from(value.replace(/^0x/, ''), 'hex')
  );
```

**Comment by dudiq** [comment]
tested on `auth`, `vault`, `connect` repos - all is fine, 

but founded two errors after install `core/utils` to `core/class`

```
\endpass-core\packages\class>yarn tsc                                                                                              yarn run v1.19.1                                                                                                                                     $ \endpass-core\packages\class\node_modules\.bin\tsc                                                                               node_modules/@endpass/utils/keystoreKeyGen.d.ts:1:60 - error TS2304: Cannot find name 'V3Keystore'.                                                                                                                                                                                                       1 export function getPublicKey(password: string, v3Keystore: V3Keystore): Buffer;                                                                                                                                 ~~~~~~~~~~                                                                                                                                                                                                                                   node_modules/@endpass/utils/keystoreKeyGen.d.ts:2:61 - error TS2304: Cannot find name 'V3Keystore'.                                                                                                                                                                                                       2 export function getPrivateKey(password: string, v3Keystore: V3Keystore): Buffer;                                                                                                                                 ~~~~~~~~~~                                                                                                                                                                                                                                                                                                                                                                                       Found 2 errors.                                                                                                                                                                                                                                                                                           error Command failed with exit code 1.                                                                                                               info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.                
```

File: packages/utils/src/keystoreKeyGen.js [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
-import keystoreCrypto from './keystoreCrypto';
+const secp256k1 = require('secp256k1');
+const keystoreCrypto = require('./keystoreCrypto');
 
-export default {
-  /**
-   * Generate public key
-   *
-   * @param password
-   * @param v3Keystore
```
Comment (dudiq) [inline packages/utils/src/keystoreKeyGen.js]: tested on `auth`, `vault`, `connect` repos - all is fine, 

but founded two errors after install `core/utils` to `core/class`

```bash
\endpass-core\packages\class>yarn tsc                                                                                              
yarn run v1.19.1                                                                                                                                     $ 
\endpass-core\packages\class\node_modules\.bin\tsc                                                                               
node_modules/@endpass/utils/keystoreKeyGen.d.ts:1:60 - error TS2304: Cannot find name 'V3Keystore'.                                                                                                                                                                                                       
1 export function getPublicKey(password: string, v3Keystore: V3Keystore): Buffer;                                                                                                                                 
~~~~~~~~~~                                                                                                                                                                                                                                   
node_modules/@endpass/utils/keystoreKeyGen.d.ts:2:61 - error TS2304: Cannot find name 
'V3Keystore'.                                                                                                                                                                                                       
2 export function getPrivateKey(password: string, v3Keystore: V3Keystore): Buffer;                                                                                                                                 
~~~~~~~~~~                                                                                                                                                                                                                                                                                                                                                                                       
Found 2 errors.                                                                                                                                                                                                                                                                                           
error Command failed with exit code 1.                                                                                                               
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.                
```

File: packages/utils/src/keystoreKeyVerify.js (hunk 1 above)
Comment (lamartire) [reply packages/utils/src/keystoreKeyVerify.js]: `prettier`

File: packages/utils/src/keystoreCrypto.js [hunk 3]
```diff
@@ -1,67 +1,64 @@
...
+const keythereum = require('keythereum');
+const { KDF_ENCRYPT_OPTIONS } = require('./constants');
+const isV3 = require('./isV3');
 
 // Monkey patch keythereum to skip generating address for private keys
 // This allows us to encrypt private keys of arbitrary length, and
 // conforms better to the Ethereum keystore V3 spec, which does not include
 // the address for privacy reasons.
 // See https://github.com/ethereum/wiki/wiki/Web3-Secret-Storage-Definition#alterations-from-version-1
 // eslint-disable-next-line
```
[the remark this answers is not in the document]
Comment (lamartire) [reply packages/utils/src/keystoreCrypto.js]: Not relates to current pull request, and to be honestly – I don't remember :)

**Pull request closed without merging**